plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/XmlSourceValidator.java

```diff
@@ -56,12 +57,22 @@
  */
 public class XmlSourceValidator implements ISourceValidator, IValidator, IExecutableExtension {
 
-  private static final Logger logger = Logger.getLogger(
-      XmlSourceValidator.class.getName());
+  private static final Logger logger = Logger.getLogger(XmlSourceValidator.class.getName());
 
   private IDocument document;
   private XmlValidationHelper helper;
 
+  private final Predicate<IFacetedProject> hasAppEngineStandardFacet;
+
+  public XmlSourceValidator() {
+    this(AppEngineStandardFacet::hasFacet);
+  }
+
+  @VisibleForTesting
+  XmlSourceValidator(Predicate<IFacetedProject> hasAppEngineStandardFacet) {
```

Comment:
What's the advantage of adding this? It seems more complex without adding functionality. The old code did not need visible for testing.

Reply:
Yeah, it's a trade-off between less code vs. less time to run tests, by allowing mocking some underlying component. Since this change isn't the matter of whether we can write tests or not, I'm not pushing hard for this change. So, if you don't like this kind of change, I'll probably drop it.